include/envoy/upstream/upstream.h

```diff
@@ -682,6 +682,11 @@ class ClusterInfo {
    */
   virtual const absl::optional<std::chrono::milliseconds> idleTimeout() const PURE;
 
+  /**
+   * @return optional maximum connection duration timeout for upstream connection pool connections.
+   */
+  virtual const absl::optional<std::chrono::milliseconds> maxConnectionDuration() const PURE;
```

Comment:
If it's not implemented for upstream connections why was this added? Can it be removed?

Reply:
removed.